cdci_data_analysis/configurer.py

```diff
@@ -33,89 +33,97 @@
+        conf = kwargs.copy()
 
-        if dispatcher_mnt_point is None:
-            logger.warning(
-                f"problem constructing {self}: dispatcher_mnt_point is None: {dispatcher_mnt_point}")
+        logger.info("building config from %s", conf)
 
-        self.data_server_url = data_server_url
 
-        # dummy prods local cache
-        self.dummy_cache = dummy_cache
+        try:
+            self.__setattr__('data_server_url', conf.pop('data_server_url'))
+            required_keys.remove('data_server_url')
+        except KeyError as e:
+            logger.error(
+                f"problem constructing {self}: data_server_url configuration key is required")
+            raise e
 
-        # path to dataserver cache
-        self.data_server_remote_path = data_server_remote_cache
+        # special cases (INTEGRAL specific)
+        # NOTE: these are required for integral, so need to be processed before for-cycle for the other required ones
+        if conf.get('data_server_remote_cache', None) is not None:
+            # path to dataserver cache
+            self.data_server_remote_path = conf.pop('data_server_remote_cache')
+        else:
+            self.data_server_remote_path = None
 
-        if dispatcher_mnt_point is not None:
-            self.dispatcher_mnt_point = os.path.abspath(dispatcher_mnt_point)
+        if conf.get('dispatcher_mnt_point', None) is not None:
+            self.dispatcher_mnt_point = os.path.abspath(conf.pop('dispatcher_mnt_point'))
             FilePath(file_dir=self.dispatcher_mnt_point).mkdir()
         else:
             self.dispatcher_mnt_point = None
 
-        self.products_url = products_url
-
         if self.dispatcher_mnt_point is not None and self.data_server_remote_path is not None:
-            #self.data_server_cache = os.path.join(self.dispatcher_mnt_point, self.data_server_remote_path)
             self.data_server_cache = os.path.join(
                 self.dispatcher_mnt_point, self.data_server_remote_path)
         else:
             self.data_server_cache = None
 
+
+        for key in required_keys:
+            try:
+                value = conf.pop(key)
+                if value is None:
+                    if os.environ.get('DISPATCHER_DEBUG_MODE', 'yes') != 'yes':
+                        logger.error(
+                            f"required configuration key {key} is None")
+                        raise ValueError(
+                            f"None value of the required configuration key {key} is only allowed in debug mode")
+                    logger.warning(
+                        f"required configuration key {key} is None")
+                self.__setattr__(key, value)
+            except KeyError as e:
+                logger.error(
+                    f"problem constructing {self}: {key} configuration key is required")
+                raise e
+
+        if conf.pop('data_server_port', None) is not None:
```

Comment:
minor, but why not make a list of two obsolete keys and slighly reduce code redundancy?

Reply:
Of course. Will do it